generators/app/prompts.js

```diff
@@ -216,3 +217,50 @@ function askModulesToBeInstalled(done, generator) {
         }
     );
 }
+
+function askForJavaVersion() {
+    if (this.existingProject) {
+        return;
+    }
+
+    const choices = [{ name: 'Java 11', value: '11' }, { name: 'Java 8', value: '1.8' }];
+    const defaultChoice = [{ name: 'Java 11', value: '11' }];
+    const PROMPT = {
+        type: 'list',
+        name: 'javaVersion',
+        message: 'Which Java version would you like to use?',
+        choices,
+        default: defaultChoice
+    };
+
+    const done = this.async();
+
+    this.prompt(PROMPT).then(prompt => {
+        this.javaVersion = prompt.javaVersion;
+        if (prompt.javaVersion === '11') {
+            askJacksonBlackBirdToBeInstalled(done, this);
```

Comment:
Shouldn't this be a filter in the blackbird "validate" prompt ?

Reply:
@cbornet : Sorry, I am not sure I understand you here. You mean the the prompt for java version should go after the prompt for blackbird prompt?

Reply:
Sorry I should have said in the "when" field of blackbird prompt, test the Java version (so after the Java version prompt).

Reply:
@cbornet : Thanks for the clarification. Now understand what you mean; I haven't done any yeoman generator development and just now realized that ```Inquirer.js``` provides the ```when``` attribute. 

Refactored the code; let me know if you see any further issues. 😄